### PR TITLE
Uniques: Reset approved account after transfer 

### PR DESCRIPTION
**Pull request by Szegoo**
Currently, the `approved` account of an item doesn't get reset to `None` after each transfer. This can cause a problem because it is possible for the owner of the item to approve an item to his other account and after transferring the item he can claim it back.

### CODE REVIEW

File: frame/uniques/src/tests.rs [hunk 1]
```diff
@@ -557,6 +557,20 @@ fn approval_lifecycle_works() {
...
+fn approved_account_gets_reset_after_transfer() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Uniques::force_create(Origin::root(), 0, 1, true));
+		assert_ok!(Uniques::mint(Origin::signed(1), 0, 42, 2));
+
+		assert_ok!(Uniques::approve_transfer(Origin::signed(2), 0, 42, 3));
+		assert_ok!(Uniques::transfer(Origin::signed(2), 0, 42, 5));
+
+		// this shouldn't work because we have just transfered the item to another account.
+		assert_noop!(Uniques::transfer(Origin::signed(3), 0, 42, 4), Error::<Test>::NoPermission);
```
Comment (ggwpez) [inline frame/uniques/src/tests.rs:570]:
```suggestion
		assert_noop!(Uniques::transfer(Origin::signed(3), 0, 42, 4), Error::<Test>::NoPermission);
		// The new owner can transfer fine:
		assert_ok!(Uniques::transfer(Origin::signed(5), 0, 42, 6));
```

**Review by shawntabrizi — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

File: frame/uniques/src/tests.rs [hunk 2]
```diff
@@ -557,6 +557,20 @@ fn approval_lifecycle_works() {
...
 }
 
+#[test]
+fn approved_account_gets_reset_after_transfer() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Uniques::force_create(Origin::root(), 0, 1, true));
+		assert_ok!(Uniques::mint(Origin::signed(1), 0, 42, 2));
+
+		assert_ok!(Uniques::approve_transfer(Origin::signed(2), 0, 42, 3));
+		assert_ok!(Uniques::transfer(Origin::signed(2), 0, 42, 5));
```
Comment (bkchr) [inline frame/uniques/src/tests.rs:567]: You should test here `buy_item` as well. `transfer` was already save before.

**Review by bkchr — APPROVED** [review]

File: frame/uniques/src/lib.rs [hunk 3]
```diff
@@ -914,6 +916,8 @@ pub mod pallet {
 		/// - `item`: The item of the item to be approved for delegated transfer.
 		/// - `delegate`: The account to delegate permission to transfer the item.
 		///
+		/// Important NOTE: The `approved` account gets resest after each transfer.
```
Comment (KiChjang) [inline frame/uniques/src/lib.rs:919]:
```suggestion
		/// Important NOTE: The `approved` account gets reset after each transfer.
```

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
@Szegoo if you would like a tip, please include your polkadot address in your PR description

**Comment by Szegoo** [comment]
> @Szegoo if you would like a tip, please include your polkadot address in your PR description

I have added my address, thanks.

**Comment by Szegoo** [comment]
> @Szegoo if you would like a tip, please include your polkadot address in your PR description

@shawntabrizi Is this tip still available :), sorry for reminding you like this.

**Pull request merged**